cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -164,28 +205,31 @@ private PullAppResponse<?> pullSingle(RepositoryManager repositoryManager, Strin
     }
   }
 
-  /**
-   * Atomic operation of writing application and push, return the push response.
-   *
-   * @param repositoryManager {@link RepositoryManager} to conduct git operations
-   * @param appToPush         {@link ApplicationDetail} to push
-   * @param commitDetails     {@link CommitMeta} from user input
-   * @return {@link PushAppResponse}
-   * @throws NoChangesToPushException if there's no change between the application in namespace and git repository
-   * @throws SourceControlException   for failures while writing config file or doing git operations
-   */
-  private PushAppResponse writeAppDetailAndPush(RepositoryManager repositoryManager,
-                                                ApplicationDetail appToPush,
-                                                CommitMeta commitDetails)
-    throws NoChangesToPushException {
+  private void cloneAndCreateBaseDir(RepositoryManager repositoryManager) {
+    try {
+      repositoryManager.cloneRemote();
+    } catch (GitAPIException | IOException e) {
+      throw new GitOperationException(String.format("Failed to clone remote repository: %s",
+          e.getMessage()), e);
```

Comment:
Typically when wrapping an exception, you don't use the e.getMessage() to form the wrapping message, otherwise it just result it duplicated message in the stacktrace.

Reply:
updated.